tools/tune/tdfb/bf_design.m

```diff
@@ -119,6 +123,37 @@
 			+(bf.mic_y(n) - bf.mic_y(m))^2 ...
 			+(bf.mic_z(n) - bf.mic_z(m))^2);
             Gamma_vv(:,n,m) = sinc(2*pi*f*lnm/bf.c);
+            if isequal(DEBUG_DIAG_LOAD_ON,1)
+                % singular value decomposition
+                Gamma_vv(:,n,m) = svd(sinc(2*pi*f*lnm/bf.c), 'econ');
+            elseif isequal(DEBUG_DIAG_LOAD_ON,2)
+                %Power Spectral Density estimate via Welch's method
+                if exist('OCTAVE_VERSION', 'builtin')
+                    [Pxyz,freq] = pwelch(sinc(2*pi*f*lnm/bf.c),[], [], N, fs, 0);
+                    Gamma_vv(:,n,m) = pwelch(Pxyz  ,[], [], N, fs, 0);
+                else
+                    [Pxyz,freq] = pwelch(2*pi*f*lnm/bf.c,hamming(510), [], N, fs);
+                    Gamma_vv(:,n,m) = pwelch(Pxyz  ,hamming(510), [], N, fs);
+                end
+            elseif isequal(DEBUG_DIAG_LOAD_ON,3)  % V is beamformer matrix
+                %Power Spectral Density estimate via Welch's method -centered two-sided PSD
```

Comment:
this code block is duplicated below, please either refactor or turn into a function

Reply:
Thanks! I have removed  the code section since I didn't observe improvement in directive index and with white noise gain.